pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -98,88 +105,218 @@ func (r *MachineSetSyncReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	case machinev1beta1.MachineAuthorityClusterAPI:
-		return r.reconcileCAPIMachineSettoMAPIMachineSet(ctx, capiMachineSet, mapiMachineSet)
+		return r.reconcileCAPIMachineSetToMAPIMachineSet(ctx, capiMachineSet, mapiMachineSet)
 	case machinev1beta1.MachineAuthorityMigrating:
-		logger.Info("machine currently migrating", "machine", mapiMachineSet.GetName())
+		logger.Info("machine set is currently migrating", "machine set", mapiMachineSet.GetName())
 		return ctrl.Result{}, nil
 	default:
-		logger.Info("machine AuthoritativeAPI has unexpected value", "AuthoritativeAPI", mapiMachineSet.Status.AuthoritativeAPI)
+		logger.Info("Unexpected value for AuthoritativeAPI", "AuthoritativeAPI", mapiMachineSet.Status.AuthoritativeAPI)
 		return ctrl.Result{}, nil
 	}
 }
 
-// reconcileCAPIMachineSettoMAPIMachineSet reconciles a CAPI MachineSet to a MAPI MachineSet.
-func (r *MachineSetSyncReconciler) reconcileCAPIMachineSettoMAPIMachineSet(ctx context.Context, capiMachineSet *capiv1beta1.MachineSet, mapiMachineSet *machinev1beta1.MachineSet) (ctrl.Result, error) {
+// reconcileMAPIMachineSetToCAPIMachineSet reconciles a MAPI MachineSet to a CAPI MachineSet.
+func (r *MachineSetSyncReconciler) reconcileMAPIMachineSetToCAPIMachineSet(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, capiMachineSet *capiv1beta1.MachineSet) (ctrl.Result, error) {
+	// This function is currently a placeholder.
 	return ctrl.Result{}, nil
 }
 
-// reconcileMAPIMachineSettoCAPIMachineSet MAPI MachineSet to a CAPI MachineSet.
-func (r *MachineSetSyncReconciler) reconcileMAPIMachineSettoCAPIMachineSet(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, capiMachineSet *capiv1beta1.MachineSet) (ctrl.Result, error) {
+// reconcileCAPIMachineSetToMAPIMachineSet reconciles a CAPI MachineSet to a
+// MAPI MachineSet.
+//
+// TODO: Platform specific implementation (currently this works only for AWS,
+// we want a switch on platform somewhere).
+// TODO: Put Gets() for Cluster + Template in helper func.
+func (r *MachineSetSyncReconciler) reconcileCAPIMachineSetToMAPIMachineSet(ctx context.Context, capiMachineSet *capiv1beta1.MachineSet, mapiMachineSet *machinev1beta1.MachineSet) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithValues("function", "reconcileCAPIMachineSetToMAPIMachineSet")
+
+	// Get the AWSCluster
+	cluster := &awscapiv1beta1.AWSCluster{}
+	clusterNamespacedName := client.ObjectKey{
+		Namespace: capiMachineSet.GetNamespace(),
+		Name:      capiMachineSet.Spec.ClusterName,
+	}
+
+	if err := r.Get(ctx, clusterNamespacedName, cluster); err != nil {
+		logger.Error(err, "Failed to get CAPI cluster", "cluster", clusterNamespacedName.Name)
+
+		r.updateSynchronizedConditionWithPatch(
+			ctx, mapiMachineSet, corev1.ConditionFalse,
+			"FailedToGetCluster", err.Error(),
+		)
+
+		return ctrl.Result{}, fmt.Errorf("failed to get CAPI cluster: %w", err)
+	}
+
+	// Get the AWSMachineTemplate
```

Comment:
```suggestion
	// Get the InfraMachineTemplate.
```

Reply:
We could stay generic with the comments, so if we plug in generic types for these, we don't need to change them then.

Reply:
Hmm. I'd not been thinking of using generics here when we need to add support for other platforms, although I've not given it a tonne of thought. 

I was thinking we'd have a library and a switch statement on platform that loads the correct helper/ reconcile func. In this instance, I think keeping the errors and comments provider specific make sense (i.e this code will ultimately live in `/lib/aws` or some such)

 Do we think generics would be a better approach?

Reply:
I don't think Dam necessarily meant generics as in the concept in computing, but more, keeping the code generic enough since you won't always have AWS specific types here

Reply:
Ah yep - we had a sync on this yesterday, I completely misunderstood dam's comment here! 🤦 

I think we're aligned now though!